golay23.h required

the latest commit requires
```
:info:build /opt/local/var/macports/build/ports_science_freedv-gui/freedv-gui/work/freedv-gui-ee194ba1eb944c2de17b0c829e0b6ee53b109c8e/src/callsign_encoder.h:28:10: fatal error: 'golay23.h' file not found
:info:build #include "golay23.h"
:info:build          ^~~~~~~~~~~
```

--- a/src/callsign_encoder.cpp
+++ b/src/callsign_encoder.cpp
@@ -24,6 +24,11 @@
 #include <cstring>
 #include "callsign_encoder.h"
 
+extern "C" {
+    extern int  golay23_encode(int data);
+    extern int  golay23_decode(int received_codeword);
+}
+
 CallsignEncoder::CallsignEncoder()
 {
     memset(&translatedCallsign_, 0, MAX_CALLSIGN);

--- a/src/callsign_encoder.h
+++ b/src/callsign_encoder.h
@@ -25,7 +25,6 @@
 
 #include <deque>
 #include "fdmdv2_defines.h"
-#include "golay23.h"
 
 class CallsignEncoder
 {

--- a/src/fdmdv2_main.cpp
+++ b/src/fdmdv2_main.cpp
@@ -24,7 +24,6 @@
 #include <deque>
 #include "fdmdv2_main.h"
 #include "osx_interface.h"
-#include "golay23.h"
 #include "callsign_encoder.h"
 
 #define wxUSE_FILEDLG   1
@@ -33,6 +32,10 @@
 #define wxUSE_GIF       1
 #define wxUSE_PCX       1
 #define wxUSE_LIBTIFF   1
+
+extern "C" {
+    extern void golay23_init(void);
+}
 
 //-------------------------------------------------------------------
 // Bunch of globals used for communication with sound card call

